avocado/plugins/xunit.py

```diff
@@ -90,6 +90,8 @@ def _render(self, result):
                 pass
             elif status == 'SKIP':
                 testcase.appendChild(Element('skipped'))
+            elif status == 'CANCEL':
+                testcase.appendChild(Element('cancelled'))
```

Comment:
This generates a

```
   <canceled/>
```

Element inside the `<testcase>`.

Which does not seem to be a valid element for that schema.  A hackish change to the xunit test shows that:

```diff
diff --git a/selftests/unit/test_xunit.py b/selftests/unit/test_xunit.py
index 64c5b60..fc6cfee 100644
--- a/selftests/unit/test_xunit.py
+++ b/selftests/unit/test_xunit.py
@@ -39,10 +39,15 @@ class xUnitSucceedTest(unittest.TestCase):
         args.xunit_output = self.tmpfile[1]
         self.job = job.Job(args)
         self.test_result = Result(FakeJob(args))
-        self.test_result.tests_total = 1
-        self.test1 = SimpleTest(job=self.job, base_logdir=self.tmpdir)
+        self.test_result.tests_total = 2
+        self.test1 = SimpleTest(job=self.job, base_logdir=os.path.join(self.tmpdir,
+                                                                       '1-test'))
         self.test1._Test__status = 'PASS'
         self.test1.time_elapsed = 1.23
+        self.test2 = SimpleTest(job=self.job, base_logdir=os.path.join(self.tmpdir,
+                                                                       '2-test'))
+        self.test2._Test__status = 'CANCEL'
+        self.test2.time_elapsed = 3.21
         unittests_path = os.path.dirname(os.path.abspath(__file__))
         self.junit_schema_path = os.path.join(unittests_path, 'junit-4.xsd')
 
@@ -54,6 +59,8 @@ class xUnitSucceedTest(unittest.TestCase):
     def testAddSuccess(self):
         self.test_result.start_test(self.test1)
         self.test_result.end_test(self.test1.get_state())
+        self.test_result.start_test(self.test2)
+        self.test_result.end_test(self.test2.get_state())
         self.test_result.end_tests()
         xunit_result = xunit.XUnitResult()
         xunit_result.render(self.test_result, self.job)
@@ -65,7 +72,7 @@ class xUnitSucceedTest(unittest.TestCase):
             raise ParseXMLError("Error parsing XML: '%s'.\nXML Contents:\n%s" % (details, xml))
         self.assertTrue(dom)
         els = dom.getElementsByTagName('testcase')
-        self.assertEqual(len(els), 1)
+        self.assertEqual(len(els), 2)
 
         with open(self.junit_schema_path, 'r') as f:
             xmlschema = etree.XMLSchema(etree.parse(f))
```

And executed:

```
======================================================================
FAIL: testAddSuccess (selftests.unit.test_xunit.xUnitSucceedTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/cleber/src/avocado/avocado/selftests/unit/test_xunit.py", line 81, in testAddSuccess
    (self.junit_schema_path, xml))
AssertionError: Failed to validate against /home/cleber/src/avocado/avocado/selftests/unit/junit-4.xsd, content:
<?xml version="1.0" encoding="UTF-8"?>
<testsuite errors="0" failures="0" name="avocado" skipped="2" tests="2" time="4.44" timestamp="2017-03-10 08:57:33.103356">
        <testcase classname="SimpleTest" name="0-SimpleTest" time="1.23"/>
        <testcase classname="SimpleTest" name="0-SimpleTest" time="3.21">
                <cancelled/>
        </testcase>
</testsuite>

```